From cb9a54169b72375a497c6f2800322f0b138022a8 Mon Sep 17 00:00:00 2001
From: Hou Qi <qi.hou@nxp.com>
Date: Fri, 6 Feb 2026 05:52:41 +0000
Subject: [PATCH] v4l2: Fix bufferpool object leak in video encoder

- In enc_handle_frame(), if the task is in paused/stopped state,
  the bufferpool object that we get from v4l2output object is not
  getting unref if the pool is set active by upstream element.

- Fix is backported from v1.24.8
  https://gitlab.freedesktop.org/gstreamer/gstreamer/-/commit/cb9a54169b72375a497c6f2800322f0b138022a8

Upstream-Status: Backport [1.24.8]

Signed-off-by: Pratik Pachange <ppachang@qti.qualcomm.com>
---
 sys/v4l2/gstv4l2videoenc.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sys/v4l2/gstv4l2videoenc.c b/sys/v4l2/gstv4l2videoenc.c
index 5d20086..5856f75 100644
--- a/sys/v4l2/gstv4l2videoenc.c
+++ b/sys/v4l2/gstv4l2videoenc.c
@@ -862,9 +862,9 @@ gst_v4l2_video_enc_handle_frame (GstVideoEncoder * encoder,
           gst_object_unref (opool);
         goto activate_failed;
       }
-      if (opool)
-        gst_object_unref (opool);
     }
+    if (opool)
+      gst_object_unref (opool);
   }
 
   if (task_state == GST_TASK_STOPPED || task_state == GST_TASK_PAUSED) {
-- 
2.43.0

